state/machine_test.go

```diff
@@ -2599,3 +2599,12 @@ func (s *MachineSuite) TestUpdateMachineSeriesSubordinateListChangeIncompatibleS
 	c.Assert(err, jc.Satisfies, state.IsIncompatibleSeriesError)
 	s.assertMachineAndUnitSeriesChanged(c, mach, "precise")
 }
+
+func (s *MachineSuite) TestCreateUgradeSeriesPrepareLock(c *gc.C) {
```

Comment:
You need a test to make sure the retry works correctly and one to fail if locked.  state.SetTestHooks() will help with that.

Reply:
@hmlanigan Perhaps you can help me with this some. Will ping you when I get a chance.

Reply:
There are some examples in state/machine_test.go: TestUpdateMachineSeries*.  Primarily I made use of Before() to check the asserts in the transactions.

Reply:
@hmlanigan I'm not sure if this test should take into account what the transaction is doing. The retry logic and number of retries is implemented in the transaction runner which is beyond the scope of this test. I'm interested to know the advantage of using the Before and After methods in this case.